src/OpenToolkit.Mathematics/Geometry/Box2.cs

```diff
@@ -19,140 +19,264 @@ namespace OpenToolkit.Mathematics
-        public static Box2 FromTLRB(float top, float left, float right, float bottom)
+        /// <param name="other">The box to query.</param>
+        /// <returns>Whether this box contains the other box</returns>
+        public bool Contains(Box2 other)
         {
-            return new Box2(left, top, right, bottom);
+            return Contains(other._max) && Contains(other._max);
         }
 
         /// <summary>
-        /// Creates a new Box2 with the specified dimensions.
+        /// Returns the distance between the nearest edge and the specified point.
         /// </summary>
-        /// <param name="left">The position of the left boundary.</param>
-        /// <param name="top">The position of the top boundary.</param>
-        /// <param name="width">The width of the box.</param>
-        /// <param name="height">The height of the box.</param>
-        /// <returns>A new OpenToolkit.Box2 with the specfied dimensions.</returns>
-        public static Box2 FromDimensions(float left, float top, float width, float height)
+        /// <param name="point">The point to find distance for.</param>
+        /// <returns>The distance between the specified point and the nearest edge.</returns>
+        public float DistanceToNearestEdge(Vector2 point)
         {
-            return new Box2(left, top, left + width, top + height);
+            var distMin = _min - point;
+            var distMax = point - _max;
+            var dist = new Vector2(MathHelper.Min(distMin.X, distMax.X), MathHelper.Min(distMin.Y, distMax.Y));
+            return dist.Length;
         }
 
         /// <summary>
-        /// Creates a new Box2 with the specified dimensions.
+        /// Returns the distance between the center of this box and the specified point.
         /// </summary>
-        /// <param name="position">The position of the top left corner.</param>
-        /// <param name="size">The size of the box.</param>
-        /// <returns>A new OpenToolkit.Box2 with the specfied dimensions.</returns>
-        public static Box2 FromDimensions(Vector2 position, Vector2 size)
+        /// <param name="point">The point to find distance for.</param>
+        /// <returns>The distance between the specified point and the center.</returns>
+        public float DistanceToCenter(Vector2 point)
         {
-            return FromDimensions(position.X, position.Y, size.X, size.Y);
+            return Vector2.Distance(point, Center);
         }
 
         /// <summary>
-        /// Gets a float describing the width of the Box2 structure.
+        /// Returns a Box2 translated by the given amount.
         /// </summary>
-        public float Width => Math.Abs(Right - Left);
+        /// <param name="distance">The distance to translate the box.</param>
+        /// <returns>The translated box.</returns>
+        public Box2 Translated(Vector2 distance)
+        {
+            return new Box2(Min + distance, Max + distance);
```

Comment:
Don't duplicate the implementation here. Copy the box, call `Translate` method on it, then return the copy.

Reply:
you could do this = Translated(...) as well though :D